lib/src/misc/frame_constraint.dart

```diff
@@ -0,0 +1,102 @@
+import 'dart:math' as math;
+
+import 'package:flutter_map/src/geo/latlng_bounds.dart';
+import 'package:flutter_map/src/map/flutter_map_frame.dart';
+import 'package:flutter_map/src/misc/point.dart';
+import 'package:latlong2/latlong.dart';
+
+abstract class FrameConstraint {
+  const FrameConstraint();
+
+  const factory FrameConstraint.unconstrained() = UnconstrainedFrame._;
```

Comment:
Would a user ever manually specify this?

Reply:
Not that I can think of. I created it as the default value to avoid needing to write special-case code that handles nulls.